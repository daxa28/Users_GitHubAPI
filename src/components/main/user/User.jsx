import React from "react";
import "./user.css";
import { NavLink } from "react-router-dom";

function User(props) {
  const user = props.user;

  return (
    <div className="user">
      <NavLink style={{ textDecoration: "none" }} to={`/card/${user.login}`}>
        <img src={user.avatar_url} alt="avatar" />
        <h4 className="user-login">{user.login}</h4>
      </NavLink>
    </div>
  );
}

export default User;
