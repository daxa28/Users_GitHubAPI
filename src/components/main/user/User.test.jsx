import { render, screen } from "@testing-library/react";

import User from "./User";
import { BrowserRouter, Route, Routes } from "react-router-dom";

const user = {
  login: "dasdfdvc",
  avatar_url:
    "https://secure.gravatar.com/avatar/25c7c18223fb42a4c6ae1c8db6f50f9b?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-user-420.png",
};

describe("User component", () => {
  it("User render", () => {
    render(
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<User user={user} />} exact />
        </Routes>
      </BrowserRouter>
    );

    expect(screen.getByRole("img")).toBeInTheDocument();
    expect(screen.getByText("dasdfdvc")).toBeInTheDocument();
  });

  it("User snapshot", () => {
    const userSnap = render(
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<User user={user} />} exact />
        </Routes>
      </BrowserRouter>
    );

    expect(userSnap).toMatchSnapshot();
  });
});
