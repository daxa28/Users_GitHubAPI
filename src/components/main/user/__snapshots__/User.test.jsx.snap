// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component User snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="user"
      >
        <a
          class=""
          href="/card/dasdfdvc"
          style="text-decoration: none;"
        >
          <img
            alt="avatar"
            src="https://secure.gravatar.com/avatar/25c7c18223fb42a4c6ae1c8db6f50f9b?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-user-420.png"
          />
          <h4
            class="user-login"
          >
            dasdfdvc
          </h4>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="user"
    >
      <a
        class=""
        href="/card/dasdfdvc"
        style="text-decoration: none;"
      >
        <img
          alt="avatar"
          src="https://secure.gravatar.com/avatar/25c7c18223fb42a4c6ae1c8db6f50f9b?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-user-420.png"
        />
        <h4
          class="user-login"
        >
          dasdfdvc
        </h4>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
